### PR TITLE
Move more functionality to the indexeddb backend

### PR DESCRIPTION
**Pull request by dbkr**
Now the backend drives the sync accumulator as well. Also moves
the backend out to a separate file.

### CODE REVIEW

File: src/store/indexeddb-local-backend.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+ *
+ * Construct a new Indexed Database store backend. This requires a call to
+ * <code>connect()</code> before this store can be used.
+ * @constructor
+ * @param {Object} indexedDBInterface The Indexed DB interface e.g
+ * <code>window.indexedDB</code>
+ * @param {string=} dbName Optional database name. The same name must be used
+ * to open the same database.
+ */
+const LocalIndexedDBStoreBackend = function LocalIndexedDBStoreBackend(
```
Comment (Kegsay) [inline src/store/indexeddb-local-backend.js]: Why the `Local` prefix? Is there a non-local one?

**Comment by Kegsay** [comment]
Otherwise probably LGTM - There's a lot of moved lines which unfortunately makes it hard to see where things have been tweaked. Superficially looks the same.

File: src/store/indexeddb-local-backend.js (hunk 1 above)
Comment (dbkr) [inline src/store/indexeddb-local-backend.js]: Not yet, but there will be.

Comment (Kegsay) [inline src/store/indexeddb-local-backend.js]: ack

**Pull request merged**